### PR TITLE
UIU-3105: hide permissionSets from settings/users navigation pane if roles interface is present

### PR DESCRIPTION
**Pull request by aidynoJ**
## Purpose

[UIU-3105](https://folio-org.atlassian.net/browse/UIU-3105) - suppress Settings > Users > Permission sets when `roles` interface is present

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by ryandberger — APPROVED** [review]

**Review by zburke — CHANGES_REQUESTED** [review]
This works, but in Eureka, we don't have mod-permissions, so the `permissions` interface won't exist at all, right? If that's correct, lets implement like we did for [fees and fines on line 133](https://github.com/folio-org/ui-users/blob/3c5aa4d1450d9e9357762c2b516937bd33bf46be/src/settings/sections.js#L133), leveraging the filter on [SettingsPage](https://github.com/folio-org/ui-users/blob/3c5aa4d1450d9e9357762c2b516937bd33bf46be/src/settings/components/SettingsPage.js#L45) to implement this in a less obtrusive/more generic way.

**Review by zburke — APPROVED** [review]
I hesitate to approve this because `getSettingSections` is just so out of place, and so specific to the `perms` route. True, it gets the job done, but without any context since there are no comments here and the route in question is not defined here. 

I think a better approach may be to handle this directly in `sections.js` where we could use the same conditionals to evaluate whether to unshift this route onto the `settingsGeneral` list. 

I also don't understand how the new tests evaluate the condition we are adding here. All the tests do is check whether the settings page renders, not whether the content it renders changes when the interface list changes.

**Comment by aidynoJ** [comment]
> I hesitate to approve this because `getSettingSections` is just so out of place, and so specific to the `perms` route. True, it gets the job done, but without any context since there are no comments here and the route in question is not defined here. 
> 
> 
> 
> I think a better approach may be to handle this directly in `sections.js` where we could use the same conditionals to evaluate whether to unshift this route onto the `settingsGeneral` list. 
> 
> 
> 
> I also don't understand how the new tests evaluate the condition we are adding here. All the tests do is check whether the settings page renders, not whether the content it renders changes when the interface list changes. 

I'll try to handle it differently.

File: src/settings/components/SectionPageItem.js [hunk 1]
```diff
@@ -38,7 +42,9 @@ SectionPageItem.propTypes = {
     label: PropTypes.element,
     perm: PropTypes.string,
     route: PropTypes.string,
-  })
+    dependsOnNoneInterface: PropTypes.string
+  }),
+  hasInterface: PropTypes.func
```
Comment (zburke) [inline src/settings/components/SectionPageItem.js:47]: Petty: when a list is sorted alphabetically, add new entries in order.

File: src/settings/components/SectionPageItem.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ import {
   NavListItem,
 } from '@folio/stripes/components';
 
-const SectionPageItem = ({ setting, path }) => {
+const SectionPageItem = ({ setting, path, hasInterface }) => {
```
Comment (zburke) [inline src/settings/components/SectionPageItem.js:8]: Let's avoid re-implementing `hasInterface` in `SettingsPage` and passing it as a prop. We can just use the original:
```
const stripes = useStripes();
// ...
if (stripes.hasInterface(setting?.dependsOnNoneInterface)) { ... }
```

**Review by zburke — APPROVED** [review]
Ooooh, this is very nice! What if we rename `dependsOnNoneInterface` to `unlessInterface`?

**Pull request merged**